BUILD.bazel

```diff
@@ -1454,99 +1454,362 @@ ray_cc_library(
-        "src/ray/core_worker/transport/concurrency_group_manager.cc",
-        "src/ray/core_worker/transport/dependency_resolver.cc",
-        "src/ray/core_worker/transport/normal_scheduling_queue.cc",
-        "src/ray/core_worker/transport/normal_task_submitter.cc",
-        "src/ray/core_worker/transport/out_of_order_actor_scheduling_queue.cc",
-        "src/ray/core_worker/transport/out_of_order_actor_submit_queue.cc",
-        "src/ray/core_worker/transport/scheduling_util.cc",
-        "src/ray/core_worker/transport/sequential_actor_submit_queue.cc",
-        "src/ray/core_worker/transport/task_receiver.cc",
     ],
     hdrs = [
-        "src/ray/core_worker/actor_creator.h",
-        "src/ray/core_worker/actor_handle.h",
-        "src/ray/core_worker/actor_manager.h",
-        "src/ray/core_worker/common.h",
-        "src/ray/core_worker/context.h",
         "src/ray/core_worker/core_worker.h",
         "src/ray/core_worker/core_worker_options.h",
         "src/ray/core_worker/core_worker_process.h",
-        "src/ray/core_worker/experimental_mutable_object_manager.h",
-        "src/ray/core_worker/experimental_mutable_object_provider.h",
         "src/ray/core_worker/fiber.h",
-        "src/ray/core_worker/future_resolver.h",
-        "src/ray/core_worker/generator_waiter.h",
-        "src/ray/core_worker/lease_policy.h",
-        "src/ray/core_worker/object_recovery_manager.h",
-        "src/ray/core_worker/profile_event.h",
-        "src/ray/core_worker/reference_count.h",
+        "src/ray/core_worker/transport/actor_submit_queue.h",
+        "src/ray/core_worker/transport/scheduling_queue.h",
+    ],
+    deps = [
+        ":actor_handle",
+        ":experimental_mutable_object_provider",
+        ":future_resolver",
+        ":generator_waiter",
+        ":normal_task_submitter",
+        ":object_recovery_manager",
+        ":plasma_store_provider",
+        ":profile_event",
+        "//src/ray/util:env",
+        "//src/ray/util:shared_lru",
+        "//src/ray/util:stream_redirection",
+    ],
+)
+
+ray_cc_library(
+    name = "core_worker_common",
+    srcs = ["src/ray/core_worker/common.cc"],
+    hdrs = ["src/ray/core_worker/common.h"],
+    deps = [
+        ":raylet_client_lib",
+        ":ray_common",
+    ],
+)
+
+ray_cc_library(
+    name = "core_worker_context",
+    srcs = ["src/ray/core_worker/context.cc"],
+    hdrs = ["src/ray/core_worker/context.h"],
+    deps = [
+        ":core_worker_common",
+        ":ray_common",
+        "@boost//:thread",
+    ]
+)
+
+ray_cc_library(
+    name = "actor_handle",
+    srcs = ["src/ray/core_worker/actor_handle.cc"],
+    hdrs = ["src/ray/core_worker/actor_handle.h"],
+    deps = [
+        ":core_worker_context",
+         "//src/ray/protobuf:worker_cc_proto",
+    ]
+)
+
+ray_cc_library(
+    name = "actor_manager",
+    srcs = [
+        "src/ray/core_worker/actor_manager.cc",
+    ],
+    hdrs = [
+        "src/ray/core_worker/actor_creator.h",
```

Comment:
it's suggested for bazel to place ALL files as a separate target, even if there's only one header file :)